x-pack/plugin/search-business-rules/src/main/java/org/elasticsearch/xpack/searchbusinessrules/retriever/PinnedRankDoc.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.searchbusinessrules.retriever;
+
+import org.apache.lucene.search.Explanation;
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.TransportVersions;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.search.rank.RankDoc;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class PinnedRankDoc extends RankDoc {
+    public static final String NAME = "pinned_rank_doc";
+
+    private final boolean isPinned;
+    private final String pinnedBy;
+
+    public PinnedRankDoc(int docId, float score, int shardIndex, boolean isPinned, String pinnedBy) {
+        super(docId, score, shardIndex);
+        this.isPinned = isPinned;
+        this.pinnedBy = (isPinned) ? Objects.requireNonNull(pinnedBy, "pinnedBy cannot be null when isPinned is true") : null;
+    }
+
+    public PinnedRankDoc(StreamInput in) throws IOException {
+        super(in);
+        this.isPinned = in.readBoolean();
+        this.pinnedBy = in.readOptionalString();
+    }
+
+    public boolean isPinned() {
+        return isPinned;
+    }
+
+    public String getPinnedBy() {
+        return pinnedBy;
+    }
+
+    @Override
+    public Explanation explain(Explanation[] sources, String[] queryNames) {
```

Comment:
Thanks for updating `explain` it's nice to see confirmation that the result was pinned.